packages/legend-graph/src/graphManager/AbstractPureGraphManager.ts

```diff
@@ -336,6 +336,21 @@ export abstract class AbstractPureGraphManager {
   abstract updateQuery(query: Query, graph: PureModel): Promise<Query>;
   abstract deleteQuery(queryId: string): Promise<void>;
 
+  //TODO: These methods will be removed once we have a way to `lazy` load elements into the graph
+  abstract buildCreateQueryLightGraph(
```

Comment:
As discussed, I think the better way to do this is not to expose these kinds of customizations on the interface, but rather to create a method like `buildLightGraph` which allows the passing in a function to further build the graph after building the light graph. The `light graph` refers to a graph where only indexing is done.